src/components/transactions/TxDetails/TxData/OnChainConfirmation/index.tsx

```diff
@@ -0,0 +1,76 @@
+import { Safe__factory } from '@/types/contracts'
+import { Link as MuiLink, Skeleton, Stack, Typography } from '@mui/material'
+import { type TransactionData } from '@safe-global/safe-gateway-typescript-sdk'
+import TxData from '..'
+import ErrorMessage from '@/components/tx/ErrorMessage'
+import { MethodDetails } from '../DecodedData/MethodDetails'
+
+import MethodCall from '../DecodedData/MethodCall'
+import { Divider } from '@/components/tx/DecodedTx'
+import Link from 'next/link'
+import { useCurrentChain } from '@/hooks/useChains'
+import { AppRoutes } from '@/config/routes'
+import { useGetTransactionDetailsQuery } from '@/store/api/gateway'
+import { useMemo } from 'react'
+import { skipToken } from '@reduxjs/toolkit/query'
+
+export const OnChainConfirmation = ({ data }: { data?: TransactionData }) => {
+  const chain = useCurrentChain()
+  const signedHash = useMemo(() => {
+    const safeInterface = Safe__factory.createInterface()
+    const params = data?.hexData ? safeInterface.decodeFunctionData('approveHash', data?.hexData) : undefined
+    if (!params || params.length !== 1) {
+      return
+    }
+
+    const signedHash = params[0] as string
+    return signedHash
+  }, [data?.hexData])
+
+  const { data: nestedTxDetails, error: txDetailsError } = useGetTransactionDetailsQuery(
+    signedHash && chain
+      ? {
+          chainId: chain.chainId,
+          txId: signedHash,
+        }
+      : skipToken,
+  )
+
+  return (
+    <Stack spacing={2}>
+      {data?.dataDecoded && <MethodCall contractAddress={data?.to.value} method={data.dataDecoded?.method} />}
+
+      {data?.dataDecoded && <MethodDetails data={data.dataDecoded} addressInfoIndex={data.addressInfoIndex} />}
+
+      <Divider />
+
+      <Stack spacing={2}>
+        <Typography variant="h5">Nested transaction:</Typography>
+        {nestedTxDetails ? (
+          <>
+            <TxData txDetails={nestedTxDetails} trusted imitation={false} />
+
+            {chain && data && (
+              <Link
+                href={{
+                  pathname: AppRoutes.transactions.tx,
+                  query: {
+                    safe: `${chain?.shortName}:${data?.to.value}`,
+                    id: nestedTxDetails.txId,
+                  },
+                }}
+                passHref
+              >
+                <MuiLink>Open nested transaction</MuiLink>
```

Comment:
Should we use an `ExternalLink` here so that it opens in a new tab?

Reply:
I thought about it but wasn't sure as this is the same interface and you could easily go back using the browsers back button. 
Will ask the product team.